pydss/flapper.py

```diff
@@ -130,8 +130,8 @@ def tend(self, bid_id, lot, bid,sender_id):
         require(self.mul(bid, self.ONE) >= self.mul(self.beg, self.bids[bid_id].bid),"Flapper/insufficient-increase")
 
         if (self.msg.sender != self.bids[bid_id].guy):
-            self.gem.move(self.message.sender, self.bids[bid_id].guy, self.bids[bid_id].bid);
-            self.bids[bid_id].guy = self.message.sender;
+            self.gem.move(self.msg.sender, self.bids[bid_id].guy, self.bids[bid_id].bid);
+            self.bids[bid_id].guy = self.msg.sender;
         self.gem.move(self.msg.sender,sender_id, bid - self.bids[bid_id].bid);
```

Comment:
replace this with a `self.gem.transferFrom(sender, self.ADDRESS, bid - self.bids[bid_id].bid)`